### PR TITLE
add debug information about hanging rpc

### PR DESCRIPTION
**Pull request by ydmao**

### CODE REVIEW

File: jetty-client/src/main/java/org/eclipse/jetty/client/HttpExchange.java [hunk 1]
```diff
@@ -145,6 +147,14 @@ public int getStatus()
         return _status.get();
     }
 
+    public void setDebugString(String debugString) {
+        _debugString = debugString;
+    }
+
+    public String get_debugString() {
```
Comment (aarondav) [inline jetty-client/src/main/java/org/eclipse/jetty/client/HttpExchange.java]: getDebugString?

File: jetty-client/src/main/java/org/eclipse/jetty/client/AsyncHttpConnection.java [hunk 2]
```diff
@@ -188,6 +188,14 @@ else if (_generator.isEmpty())
 
                     if (complete)
                     {
+                        HttpExchange exchangeSnapshot = _exchange;
```
Comment (aarondav) [inline jetty-client/src/main/java/org/eclipse/jetty/client/AsyncHttpConnection.java]: Could you add a comment linking to the JIRA?

File: jetty-client/src/main/java/org/eclipse/jetty/client/AsyncHttpConnection.java [hunk 3]
```diff
@@ -188,6 +188,14 @@ else if (_generator.isEmpty())
...
+                        if (exchangeSnapshot != null) {
+                            String debugString = "failed=" + failed + ",generatorComplete=" +
+                                _generator.isComplete() + ", parser=[" +
```
Comment (aarondav) [inline jetty-client/src/main/java/org/eclipse/jetty/client/AsyncHttpConnection.java]: maybe no space-after-comma, if that's what we're going for..?

File: jetty-client/src/main/java/org/eclipse/jetty/client/HttpExchange.java [hunk 4]
```diff
@@ -126,6 +126,8 @@
     boolean _onResponseCompleteDone;
     boolean _onDone; // == onConnectionFail || onException || onExpired || onCancelled || onResponseCompleted && onRequestCompleted
 
+    private String _debugString = "none";
```
Comment (aarondav) [inline jetty-client/src/main/java/org/eclipse/jetty/client/HttpExchange.java]: Could you add a comment on this guy, with JIRA too?

File: jetty-client/src/main/java/org/eclipse/jetty/client/HttpExchange.java (hunk 1 above)
Comment (ydmao) [reply jetty-client/src/main/java/org/eclipse/jetty/client/HttpExchange.java]: of course!

**Comment by ydmao** [comment]
@aarondav PTAL. thanks!

**Comment by aarondav** [comment]
LGTM

**Pull request merged**